packages/nuxt/src/app/composables/ssr.ts

```diff
@@ -132,7 +132,7 @@ export function onPrehydrate (callback: string | ((el: HTMLElement) => void), ke
 
   useServerHead({
     script: [{
-      key: vm && key ? key : code,
+      key: vm && key ? key : undefined,
```

Comment:
As Unhead v2 does not create a hash from the provided `key`, this would set the `data-hid` to the script inner content. I don't think we need a key in these instances as:
- Unhead will dedupe based on inner content when no script is provided
- This is server-only so the key may not be needed at all (key is used to hydrate without causing duplicates in this instances)